# Assignment on Measures of Central Tendency and Dispersion

In [1]:
import numpy as np
import pandas as pd


In [3]:
df=pd.read_csv('StudentsPerformance.csv')


In [4]:
df.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

# 1 count of males and female participants in the test

In [6]:
count=df['gender'].value_counts()
count

female    518
male      482
Name: gender, dtype: int64

there are 518 male participants and 482 female participants in the test

# 2 students parental level of education

In [11]:
df['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

375 parents only have school level education out of 1000 students.

# 3 Students scored avarage for math, reading and writing

# Based on gender

In [14]:
Gender_score= df[['gender', 'math score', 'reading score','writing score']].groupby('gender').mean()
Gender_score

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

# Based on test preparation course

In [17]:
TPC= df[['test preparation course', 'math score', 'reading score','writing score']].groupby('test preparation course').mean()
TPC

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

# 4 scoring variation for math, reading and writing

# Based on gender

In [22]:
gender_variation= df[['gender','math score','reading score','writing score']].groupby('gender').std()
gender_variation

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

# Based on test preparation course

In [23]:
TPC_variation= df[['test preparation course','math score','reading score','writing score']].groupby('test preparation course').std()
TPC_variation

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

# 5 bonus points to top 25% of the students based on math score

In [24]:
# out of 1000 students 25% is 250
bonus= df.nlargest(250,['math score'])
bonus

gender race/ethnicity parental level of education         lunch  \
515  female        group E                some college      standard   
516  female        group E           bachelor's degree      standard   
517  female        group E          associate's degree      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
..      ...            ...                         ...           ...   
428  female        group D                some college  free/reduced   
429  female        group E            some high school      standard   
849    male        group A           bachelor's degree      standard   
850    male        group C          associate's degree  free/reduced   
851    male        group D                some college      standard   

    test preparation course  math score  reading score  writing score  
515                    none         100             92             97  
516                    none         100            100            100  
517                    none         100            100            100  
996               completed         100             96             86  
997               completed         100             97             99  
..                      ...         ...            ...            ...  
428                    none          77             86             86  
429                    none          77             79             80  
849                    none          77             67             68  
850                    none          77             67             64  
851               completed          77             62             62  

[250 rows x 8 columns]